### ISSUE TITLE
TreeSequence Builder Utility for Tests

### ISSUE THREAD

**Issue by daniel-goldstein** [tskit-dev/tskit#652]
I've struggled a fair bit with writing and interpreting trees/tree sequences with the current methods of hand-rolling test cases. As far as I know, the options are to manually add entries to tables or parse a textual representation of tables. Both are pretty verbose and difficult to visualize imo. I feel like it would be nice to have some builder that can make tests easier to write and understand. Something like:

```
b = TreeSequenceBuilder(samples=[0, 1, 2, 3])
b.set_node_times([0, 0, 0, 0, 1, 2, 3])
b.start(0.0)
  .insert(children=[0, 1], parent=4).insert(children=[2, 3], parent=5)
  .insert(children=[4, 5], parent=6)
  .add_mutation(above=5)
  .break(1.0)
  .move(children=[0], new_parent=6)
  .move(children=[1], new_parent=5)
  .break(2.0)
  .remove(children=[0, 5], parent=6)
  .end(3.0)
```

@jeromekelleher @benjeffery Thoughts on the utility of something like this?

**Comment by jeromekelleher** [comment]
Looks interesting @daniel-goldstein, and I agree we don't have great tools for this right now. The idea here is a stateful tree-by-tree description, I think?

**Comment by daniel-goldstein** [comment]
Yes exactly. I also think there’s some constraints like how many nodes must exist, declaring populations, when edges end, etc. that can be derived from the process above instead of explicitly added to tables.
